src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -348,15 +348,15 @@ public XFormParser(Document form, Document instance) {
         _instDoc = instance;
     }
 
-    public FormDef parse() throws IOException {
+    public FormDef parse(String lastSavedSrc) throws IOException {
```

Comment:
~Could you restore the old `parse()` method to maintain backwards compatibility? It can call to `parse(String lastSavedSrc)` with a null.~ I see that you added overrides in another commit ;)